src/rulesets/oas/functions/oasDocumentSchema.ts

```diff
@@ -0,0 +1,29 @@
+import * as AJV from 'ajv';
+import { ISchemaOptions } from '../../../functions/schema';
+import { IFunction, IFunctionContext } from '../../../types';
+
+// /shrug
+function prepareResults(errors: AJV.ErrorObject[]) {
```

Comment:
@P0lip As this is about to become a somewhat cornerstone of schema validation, how about covering it with targeted unit tests to better express what we expect from this function?

Reply:
Agreed. Also more comments on _why_ would help future maintainers, our future selves (and my current self)  understand whats going on better.